### PR TITLE
Completed code challenge binary search, updated readme and included p…

### PR DESCRIPTION
**Pull request by SharinaS**
…ic of whiteboarding.
 - [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [ ] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401Challenges/src/test/java/code401Challenges/BinarySearchTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        );
+    }
+    @Test
+    public void testFindIndexOddNumberedArray() {
+        int[] arr = new int[]{4,8,15,16,23,42,56};
+        assertEquals(
+                "the answer should be 6",
+                6,
+                BinarySearch.binarySearch(arr, 56)
+        );
```
Comment (jeffborda) [inline code401Challenges/src/test/java/code401Challenges/BinarySearchTest.java]: This is a good test since it is checking the code block in the second 'else if'. Another good one would be to find the '4', which would test the first 'else if'.

File: code401Challenges/src/test/java/code401Challenges/BinarySearchTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    @Test
+    public void testCheckArraySingleItem() {
+        int[] arr = new int[]{1};
+        assertEquals(
+                "the answer should be 0",
+                0,
+                BinarySearch.binarySearch(arr, 1)
+        );
```
Comment (jeffborda) [inline code401Challenges/src/test/java/code401Challenges/BinarySearchTest.java]: These kind of edge cases are always good things to test. Nicely done.

File: code401Challenges/src/main/java/code401Challenges/BinarySearch.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package code401Challenges;
+
+// Use binary search to find the index of a given integer within an array, called key.
+// Resource: https://www.geeksforgeeks.org/binary-search-bisect-in-python/
+// Resource: Wikipedia's Binary Search article
```
Comment (jeffborda) [inline code401Challenges/src/main/java/code401Challenges/BinarySearch.java]: Excellent comments/citations, this will serve you well!

File: code401Challenges/src/main/java/code401Challenges/BinarySearch.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+public class BinarySearch {
+    public static int binarySearch(int[] arr, int key){
+        // establish the beginning and ending of the array
+        int start = 0;
+        int end = arr.length-1;
```
Comment (jeffborda) [inline code401Challenges/src/main/java/code401Challenges/BinarySearch.java]: Just a small thing... but keep your spacing consistent, i.e.   int end = arr.length - 1;
